Typescript  Property 'collectionName' does not exist on type 'Collection'.

running on Version 11.3

```
 error TS2339: Property 'collectionName' does not exist on type 'Collection'.

43         console.log(`${model.collection.collectionName}`);
                                                           ~~~~~~~~~~~~~~
```

thanks in advance.

--- a/index.d.ts
+++ b/index.d.ts
@@ -294,7 +294,11 @@ declare module "mongoose" {
   }
 
   class Collection {
+    /**The name of the collection */
     name: string;
+    
+    /**The name of the collection */
+    collectionName: string;
   }
 
   class Document {

--- a/index.d.ts
+++ b/index.d.ts
@@ -293,12 +293,46 @@ declare module "mongoose" {
     watch(pipeline?: Array<any>, options?: mongodb.ChangeStreamOptions): mongodb.ChangeStream;
   }
 
-  class Collection {
-    /**The name of the collection */
-    name: string;
-    
-    /**The name of the collection */
+   /*
+   * section collection.js
+   * http://mongoosejs.com/docs/api.html#collection-js
+   */
+  interface CollectionBase extends mongodb.Collection {
+    /*
+      * Abstract methods. Some of these are already defined on the
+      * mongodb.Collection interface so they've been commented out.
+      */
+    ensureIndex(...args: any[]): any;
+    findAndModify(...args: any[]): any;
+    getIndexes(...args: any[]): any;
+
+    /** The collection name */
     collectionName: string;
+    /** The Connection instance */
+    conn: Connection;
+    /** The collection name */
+    name: string;
+  }
+
+  /*
+   * section drivers/node-mongodb-native/collection.js
+   * http://mongoosejs.com/docs/api.html#drivers-node-mongodb-native-collection-js
+   */
+  let Collection: Collection;
+  interface Collection extends CollectionBase {
+    /**
+     * Collection constructor
+     * @param name name of the collection
+     * @param conn A MongooseConnection instance
+     * @param opts optional collection options
+     */
+    new(name: string, conn: Connection, opts?: any): Collection;
+    /** Formatter for debug print args */
+    $format(arg: any): string;
+    /** Debug print helper */
+    $print(name: any, i: any, args: any[]): void;
+    /** Retrieves information about this collections indexes. */
+    getIndexes(): any;
   }
 
   class Document {

--- a/test/typescript/collection.ts
+++ b/test/typescript/collection.ts
@@ -1,0 +1,12 @@
+import { Schema, model, Document } from 'mongoose';
+
+const schema: Schema = new Schema({ name: { type: 'String' } });
+
+interface ITest extends Document {
+  name?: string;
+  age?: number;
+}
+
+const Test = model<ITest>('Test', schema);
+
+Test.collection.collectionName;

--- a/test/typescript/collection.ts
+++ b/test/typescript/collection.ts
@@ -10,3 +10,8 @@ interface ITest extends Document {
 const Test = model<ITest>('Test', schema);
 
 Test.collection.collectionName;
+Test.collection.findOne({})
+Test.collection.findOneAndDelete({})
+Test.collection.ensureIndex()
+Test.collection.findAndModify()
+Test.collection.getIndexes()

--- a/test/typescript/main.test.js
+++ b/test/typescript/main.test.js
@@ -156,6 +156,14 @@ describe('typescript syntax', function() {
     }
     assert.equal(errors.length, 0);
   });
+
+  it('collection', function() {
+    const errors = runTest('collection.ts');
+    if (process.env.D && errors.length) {
+      console.log(errors);
+    }
+    assert.equal(errors.length, 0);
+  });
 });
 
 function runTest(file) {

--- a/test/typescript/main.test.js
+++ b/test/typescript/main.test.js
@@ -3,11 +3,7 @@
 const assert = require('assert');
 const typescript = require('typescript');
 
-const tsconfig = {
-  allowSyntheticDefaultImports: true,
-  esModuleInterop: true,
-  outDir: `${__dirname}/dist`
-};
+const tsconfig = require('./ts.config.json');
 
 describe('typescript syntax', function() {
   this.timeout(5000);

--- a/test/typescript/ts.config.json
+++ b/test/typescript/ts.config.json
@@ -1,0 +1,9 @@
+{
+  "allowSyntheticDefaultImports": true,
+  "esModuleInterop": true,
+  "outDir": "test/dist",
+  "include": [
+    "./**/*.ts",
+    "../../index.d.ts"
+  ]
+}

